crates/rpc-types-eth/src/transaction/mod.rs

```diff
@@ -319,7 +331,67 @@ impl TransactionResponse for Transaction {
     fn input(&self) -> &Bytes {
         &self.input
     }
+
+    fn gas_price(tx: &impl alloy_consensus::Transaction, base_fee: Option<u64>) -> u128 {
+        match TxType::try_from(tx.ty()).expect("should decode") {
+            TxType::Legacy | TxType::Eip2930 => tx.max_fee_per_gas(),
+            TxType::Eip1559 | TxType::Eip4844 | TxType::Eip7702 => {
+                // the gas price field for EIP1559 is set to `min(tip, gasFeeCap - baseFee) +
+                // baseFee`
+                base_fee
+                    .and_then(|base_fee| {
+                        tx.effective_tip_per_gas(base_fee).map(|tip| tip + base_fee as u128)
+                    })
+                    .unwrap_or_else(|| tx.max_fee_per_gas())
+            }
+        }
+    }
+
+    fn max_fee_per_gas(tx: impl alloy_consensus::Transaction) -> Option<u128> {
+        match TxType::try_from(tx.ty()).expect("should decode") {
+            TxType::Legacy | TxType::Eip2930 => None,
+            TxType::Eip1559 | TxType::Eip4844 | TxType::Eip7702 => Some(tx.max_fee_per_gas()),
+        }
+    }
+
+    fn fill(
+        signed_tx: Signed<impl alloy_consensus::Transaction>,
+        signer: Address,
+        tx_info: TransactionInfo,
+    ) -> Self {
```

Comment:
these functions won't work, because they constrain type creation, we can not lock this in in the trait function

Reply:
how come?